src/web/app/pages-logs/logs-histogram-page/logs-histogram-page.component.ts

```diff
@@ -0,0 +1,157 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import moment from 'moment-timezone';
+import { EMPTY, forkJoin, Observable } from 'rxjs';
+import { concatMap, expand, finalize, map, reduce } from 'rxjs/operators';
+import { LogService } from '../../../services/log.service';
+import { StatusMessageService } from '../../../services/status-message.service';
+import { LOCAL_DATE_TIME_FORMAT, TimeResolvingResult, TimezoneService } from '../../../services/timezone.service';
+import { ApiConst } from '../../../types/api-const';
+import { GeneralLogEntry, GeneralLogs } from '../../../types/api-output';
+import { LogsHistogramDataModel } from '../../components/logs-histogram/logs-histogram-model';
+import { DateFormat } from '../../components/session-edit-form/session-edit-form-model';
+import { TimeFormat } from '../../components/session-edit-form/time-picker/time-picker.component';
+import { ErrorMessageOutput } from '../../error-message-output';
+
+/**
+ * Model for searching of logs
+ */
+interface SearchLogsFormModel {
+  logsDateFrom: DateFormat;
+  logsDateTo: DateFormat;
+  logsTimeFrom: TimeFormat;
+  logsTimeTo: TimeFormat;
+}
+
+/**
+ * Query parameters for HTTP request
+ */
+interface QueryParams {
+  searchFrom: string;
+  searchUntil: string;
+  severity: string;
+  nextPageToken?: string;
+}
+
+/**
+ * Page for viewing logs in histogram.
+ */
+@Component({
+  selector: 'tm-logs-histogram-page',
+  templateUrl: './logs-histogram-page.component.html',
+  styleUrls: ['./logs-histogram-page.component.scss'],
+})
+export class LogsHistogramPageComponent implements OnInit {
+
+  readonly LOGS_RETENTION_PERIOD_IN_DAYS: number = ApiConst.LOGS_RETENTION_PERIOD;
+  readonly LOGS_RETENTION_PERIOD_IN_MILLISECONDS: number = this.LOGS_RETENTION_PERIOD_IN_DAYS * 24 * 60 * 60 * 1000;
+
+  formModel: SearchLogsFormModel = {
+    logsDateFrom: { year: 0, month: 0, day: 0 },
+    logsTimeFrom: { hour: 0, minute: 0 },
+    logsDateTo: { year: 0, month: 0, day: 0 },
+    logsTimeTo: { hour: 0, minute: 0 },
+  };
+  previousQueryParams: QueryParams = {
+    searchFrom: '',
+    searchUntil: '',
+    severity: '',
+  };
+  dateToday: DateFormat = { year: 0, month: 0, day: 0 };
+  earliestSearchDate: DateFormat = { year: 0, month: 0, day: 0 };
+  nextPageToken: string | undefined = '';
+  isLoading: boolean = false;
+  isSearching: boolean = false;
+  hasResult: boolean = false;
+  searchResult: LogsHistogramDataModel[] = [];
+
+  constructor(private logService: LogService,
+    private timezoneService: TimezoneService,
+    private statusMessageService: StatusMessageService,
+    private router: Router) { }
+
+  ngOnInit(): void {
```

Comment:
I'm not sure if you need an entire new page for this. Especially looking at the number of duplications between here and the main logs page.

Why not make the histogram an alternative way of viewing in the same page? The underlying data structure is essentially the same. Keep the filter at the top, add the option to display as table or as histogram. I do believe you'll find your work a lot simpler.

Reply:
I think it is possible to put them in the same page, but I feel that it might be good to keep the histogram view as a separate component since it needs to get all logs from the selected time period and the logs are processed differently. I will try putting them in the same page and reduce the duplication.

Reply:
The logs page was less complicated than I thought it would be, so I removed the histogram view component. The additional filters will be hidden when users switch to histogram.
![image](https://user-images.githubusercontent.com/58677983/126093010-4ac2cf9b-544d-40e2-8d95-4232307c8552.png)